client/signup/steps/user/index.jsx

```diff
@@ -447,6 +451,17 @@ export class UserStep extends Component {
 			isSocialSignupEnabled = true;
 		}
 
+		if ( this.props.step && 'completed' === this.props.step.status ) {
```

Comment:
Rendering doesn't seem like the right time to perform such a side effect. At the time the redirect happens, the redirect will have finished, which is likely why you're seeing the form shortly before the redirect.

Have you tried moving the redirection logic to `componentDidUpdate()`? Furthermore, it might need to be moved above to the controller level, since it happening in the step component could already be "too late".